### ISSUE TITLE
Error 500 after restart

### ISSUE THREAD

**Issue by fdhex** [CachetHQ/Docker#68]
I think this is a separate issue from the other's which symptoms appear right after you start the container with docker-compose and before you have a chance to configure anything.

After a successful setup on the latest master with APC(u) as backend (after the key:generate and app:install commands, and a first restart, otherwise it's stuck on 500), I try to restart with `docker-compose restart`, after that I get only 500 errors.

Here are the logs from ./storages/logs

```
[2016-05-12 08:12:15] production.CRITICAL: Symfony\Component\Debug\Exception\FatalThrowableError: Fatal error: Call to undefined function Illuminate\Cache\apc_store() in /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/ApcWrapper.php:45
Stack trace:
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/ApcStore.php(63): Illuminate\Cache\ApcWrapper->put('laravel9edbf40a...', 'a:1:{s:5:"flash...', 7200)
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(206): Illuminate\Cache\ApcStore->put('9edbf40addd475d...', 'a:1:{s:5:"flash...', 120)
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/CacheBasedSessionHandler.php(66): Illuminate\Cache\Repository->put('9edbf40addd475d...', 'a:1:{s:5:"flash...', 120)
#3 /var/www/html/bootstrap/cache/compiled.php(11997): Illuminate\Session\CacheBasedSessionHandler->write('9edbf40addd475d...', 'a:1:{s:5:"flash...')
#4 /var/www/html/bootstrap/cache/compiled.php(11818): Illuminate\Session\Store->save()
#5 /var/www/html/bootstrap/cache/compiled.php(2356): Illuminate\Session\Middleware\StartSession->terminate(Object(Illuminate\Http\Request), Object(Illuminate\Http\Response))
#6 /var/www/html/public/index.php(57): Illuminate\Foundation\Http\Kernel->terminate(Object(Illuminate\Http\Request), Object(Illuminate\Http\Response))
#7 {main} {"identification":{"id":"bd95a388-507c-4726-8417-292d126b2383"}}
[2016-05-12 08:12:16] production.CRITICAL: Symfony\Component\Debug\Exception\FatalThrowableError: Fatal error: Call to undefined function Illuminate\Cache\apc_fetch() in /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/ApcWrapper.php:32
Stack trace:
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/ApcStore.php(46): Illuminate\Cache\ApcWrapper->get('laravel2742ab09...')
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(129): Illuminate\Cache\ApcStore->get('2742ab0920b7477...')
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/CacheBasedSessionHandler.php(58): Illuminate\Cache\Repository->get('2742ab0920b7477...', '')
#3 /var/www/html/bootstrap/cache/compiled.php(11931): Illuminate\Session\CacheBasedSessionHandler->read('2742ab0920b7477...')
#4 /var/www/html/bootstrap/cache/compiled.php(11923): Illuminate\Session\Store->readFromHandler()
#5 /var/www/html/bootstrap/cache/compiled.php(11915): Illuminate\Session\Store->loadSession()
#6 /var/www/html/bootstrap/cache/compiled.php(11825): Illuminate\Session\Store->start()
#7 /var/www/html/bootstrap/cache/compiled.php(11804): Illuminate\Session\Middleware\StartSession->startSession(Object(Illuminate\Http\Request))
#8 [internal function]: Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#9 /var/www/html/bootstrap/cache/compiled.php(9823): call_user_func_array(Array, Array)
#10 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#12 /var/www/html/bootstrap/cache/compiled.php(12990): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#13 [internal function]: Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#14 /var/www/html/bootstrap/cache/compiled.php(9823): call_user_func_array(Array, Array)
#15 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#17 /var/www/html/bootstrap/cache/compiled.php(12927): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#18 [internal function]: Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#19 /var/www/html/bootstrap/cache/compiled.php(9823): call_user_func_array(Array, Array)
#20 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#21 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#22 [internal function]: Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#23 /var/www/html/bootstrap/cache/compiled.php(9813): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#24 /var/www/html/bootstrap/cache/compiled.php(8113): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#25 /var/www/html/bootstrap/cache/compiled.php(8104): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#26 /var/www/html/bootstrap/cache/compiled.php(8094): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#27 /var/www/html/bootstrap/cache/compiled.php(2400): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#28 [internal function]: Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(52): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#30 /var/www/html/bootstrap/cache/compiled.php(3233): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#31 [internal function]: Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#32 /var/www/html/bootstrap/cache/compiled.php(9823): call_user_func_array(Array, Array)
#33 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#35 /var/www/html/vendor/fideloper/proxy/src/TrustProxies.php(46): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#36 [internal function]: Fideloper\Proxy\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
#37 /var/www/html/bootstrap/cache/compiled.php(9823): call_user_func_array(Array, Array)
#38 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#39 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#40 [internal function]: Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#41 /var/www/html/bootstrap/cache/compiled.php(9813): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#42 /var/www/html/bootstrap/cache/compiled.php(2347): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#43 /var/www/html/bootstrap/cache/compiled.php(2331): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#44 /var/www/html/public/index.php(54): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#45 {main} {"identification":{"id":"6147ecd4-294d-4b8b-96ac-523f928493d0"}}
[2016-05-12 08:12:16] production.CRITICAL: Symfony\Component\Debug\Exception\FatalThrowableError: Fatal error: Call to undefined function Illuminate\Cache\apc_store() in /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/ApcWrapper.php:45
Stack trace:
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/ApcStore.php(63): Illuminate\Cache\ApcWrapper->put('laravel2742ab09...', 'a:1:{s:5:"flash...', 7200)
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(206): Illuminate\Cache\ApcStore->put('2742ab0920b7477...', 'a:1:{s:5:"flash...', 120)
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/CacheBasedSessionHandler.php(66): Illuminate\Cache\Repository->put('2742ab0920b7477...', 'a:1:{s:5:"flash...', 120)
#3 /var/www/html/bootstrap/cache/compiled.php(11997): Illuminate\Session\CacheBasedSessionHandler->write('2742ab0920b7477...', 'a:1:{s:5:"flash...')
#4 /var/www/html/bootstrap/cache/compiled.php(11818): Illuminate\Session\Store->save()
#5 /var/www/html/bootstrap/cache/compiled.php(2356): Illuminate\Session\Middleware\StartSession->terminate(Object(Illuminate\Http\Request), Object(Illuminate\Http\Response))
#6 /var/www/html/public/index.php(57): Illuminate\Foundation\Http\Kernel->terminate(Object(Illuminate\Http\Request), Object(Illuminate\Http\Response))
#7 {main} {"identification":{"id":"4b7e1152-ca1f-4b4b-8ada-99807553ab6b"}}
```

**Comment by fdhex** [comment]
Update: I don't even know how I got to this step, probably with some images built at other steps.

I deleted all my docker images, built using master, and now whatever I do I can't even go to the setup page. Logs below

```
[2016-05-12 08:56:01] development.ERROR: exception 'RuntimeException' with message 'No supported encrypter found. The cipher and / or key length are invalid.' in /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php:45
Stack trace:
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(25): Illuminate\Encryption\EncryptionServiceProvider->getEncrypterForKeyAndCipher('SECRET', 'AES-256-CBC')
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(735): Illuminate\Encryption\EncryptionServiceProvider->Illuminate\Encryption\{closure}(Object(Illuminate\Foundation\Application), Array)
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\Container\Container->build(Object(Closure), Array)
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\Container\Container->make('encrypter', Array)
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1178): Illuminate\Foundation\Application->make('encrypter')
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/QueueManager.php(139): Illuminate\Container\Container->offsetGet('encrypter')
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(146): Illuminate\Queue\QueueManager->connection(NULL)
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(110): Illuminate\Queue\Worker->pop(NULL, NULL, 0, '3', '3')
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(72): Illuminate\Queue\Console\WorkCommand->runWorker(NULL, NULL, 0, 128, false)
#9 [internal function]: Illuminate\Queue\Console\WorkCommand->fire()
#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(507): call_user_func_array(Array, Array)
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(169): Illuminate\Container\Container->call(Array)
#12 /var/www/html/vendor/symfony/console/Command/Command.php(256): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(155): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /var/www/html/vendor/symfony/console/Application.php(791): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /var/www/html/vendor/symfony/console/Application.php(186): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /var/www/html/vendor/symfony/console/Application.php(117): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(107): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/html/artisan(36): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 {main} {"identification":{"id":"51400427-d963-49af-b324-9d0de925e022"}}
```

**Comment by Jamie-Owen** [comment]
Should also be fixed with a combination of the 2.2.1 release and PR #71.
